src/markup/codemirror/search-plugin/view/SearchPopup.tsx

```diff
@@ -0,0 +1,183 @@
+            isWholeWord: !isWholeWord,
+        });
+        setIsWholeWord((isWholeWord) => !isWholeWord);
+        setInputFocus();
+    };
+
+    const handleSearchKeyPress: TextInputProps['onKeyPress'] = enterKeyHandler(handleNext);
+
+    return (
+        <Card className={b()}>
+            <div className={b('header')}>
+                <span className={b('title')}> {i18nSearch('title')}</span>
+                <Button onClick={handleClose} size="s" view="flat">
+                    <Icon data={Xmark} size={14} />
+                </Button>
+            </div>
+            <TextInput
+                controlRef={textInputRef}
+                className={sp({mb: 2})}
+                size="s"
+                autoFocus
+                onKeyPress={handleSearchKeyPress}
+                onChange={handleInputChange}
+                value={query}
+                endContent={
+                    <>
+                        <Button onClick={handlePrev}>
+                            <Icon data={ChevronUp} size={12} />
+                        </Button>
+                        <Button onClick={handleNext}>
+                            <Icon data={ChevronDown} size={12} />
+                        </Button>
+                    </>
+                }
+            />
+            <Checkbox
+                size="m"
+                onUpdate={handleIsCaseSensitive}
+                checked={isCaseSensitive}
+                className={sp({mr: 4})}
+            >
+                {i18nSearch('label_case-sensitive')}
+            </Checkbox>
+            <Checkbox size="m" onUpdate={handleIsWholeWord} checked={isWholeWord}>
+                {i18nSearch('label_whole-word')}
+            </Checkbox>
+        </Card>
+    );
+};
+
+export interface SearchPopupProps {
+    anchor: HTMLElement;
+    onChange: (query: string) => void;
+    onClose: () => void;
+    onSearchNext: () => void;
+    onSearchPrev: () => void;
+    onConfigChange: (config: SearchConfig) => void;
+    open: boolean;
+}
+
+export const SearchPopup: React.FC<SearchPopupProps> = ({open, anchor, onClose, ...props}) => {
+    const anchorRef = useRef<HTMLElement>(anchor);
+
+    return (
+        <>
+            {anchorRef && (
+                <Popup
+                    onEscapeKeyDown={onClose}
+                    open={open}
+                    anchorRef={anchorRef as PopoverAnchorRef}
+                    placement="bottom-end"
+                >
+                    <SearchCard onClose={onClose} {...props} />
+                </Popup>
+            )}
+        </>
+    );
+};
+
+SearchPopup.displayName = 'SearchPopup';
+
+interface PopupWithRefProps extends Omit<SearchPopupProps, 'anchor'> {
+    anchor: HTMLElement | null;
+}
+export const PortalWithPopup: React.FC<PopupWithRefProps> = ({anchor, open, onClose, ...props}) => {
+    return (
+        <>{anchor && <SearchPopup open={open} onClose={onClose} anchor={anchor} {...props} />}</>
+    );
+};
```

Comment:
You can replace this component with render function like [this](https://github.com/gravity-ui/markdown-editor/blob/9b808128a2042a91cfeead70a2d1113c4fe55961/src/markup/codemirror/files-upload-plugin/widget.tsx#L11C17-L11C29).
And remove import of React from `search-plugin/plugin.ts`

Reply:
fixed